lib/_tls_wrap.js

```diff
@@ -382,34 +398,38 @@ TLSSocket.prototype._wrapHandle = function(wrap) {
   }
 
   // Wrap socket's handle
-  var context = options.secureContext ||
-                options.credentials ||
-                tls.createSecureContext(options);
-  res = tls_wrap.wrap(handle._externalStream,
-                      context.context,
-                      !!options.isServer);
+  const context = options.secureContext ||
+                  options.credentials ||
+                  tls.createSecureContext(options);
+  const res = tls_wrap.wrap(handle._externalStream,
+                            context.context,
+                            !!options.isServer);
   res._parent = handle;
   res._parentWrap = wrap;
   res._secureContext = context;
   res.reading = handle.reading;
+  this[kRes] = res;
   Object.defineProperty(handle, 'reading', {
-    get: function get() {
-      return res.reading;
+    get: () => {
+      return this[kRes].reading;
```

Comment:
I don't think there is a need to use `this[kRes]` here, as we are closing upon res anyway.

Reply:
This was added because I think otherwise it's a cyclic reference, no? Also see this commit https://github.com/apapirovski/node/commit/f7620fb96d339f704932f9bb9a0dceb9952df2d4